rustls/src/client/mod.rs

```diff
@@ -463,12 +463,12 @@ impl Connection for ClientConnection {
         self.common.send_close_notify()
     }
 
-    fn peer_certificates(&self) -> Option<Vec<key::Certificate>> {
+    fn peer_certificates(&self) -> Option<&[key::Certificate]> {
```

Comment:
If we're hoping to create a stable API then it would be better if the return type were `Option<PeerCertificatesIterator<'a>>` where `PeerCertificatesIterator` implements `Iterator<Item=&[u8]>` and where the initial implementation is `struct PeerCertificatesIterator<'a>(&'a [u8])`.

This way, in the future we can potentially eliminate more of the copying and allocating.

Reply:
I'll change it after we decide what to do with #685; then, it could just be `impl Iterator<Item = &key::Certificate<'_>>`.